subcommand/injector/command.go

```diff
@@ -218,11 +221,12 @@ func (c *Command) Run(args []string) int {
 	}()
 	go func() {
 		select {
+		case <-exitOnError:
+			logger.Error("shutting down due to errors")
```

Comment:
As per the comment above, I think this log message should include the error that triggered the shutdown.

Reply:
It looks like on `exitOnError` does not result in a non-zero status being returned to the caller. I presumes that would make its way to `os.Exit()`.

Reply:
> It looks like on exitOnError does not result in a non-zero status being returned to the caller. I presumed that it would make its way to os.Exit().

The flow here is:
* `c.shutdownHandler()` calls `server.Shutdown()`
* `server.Shutdown()` causes `server.ListenAndServeTLS()` to return an error, triggering a `return 1` for this function:
https://github.com/hashicorp/vault-k8s/blob/b73527fa89b29ec705daf58e49b03bd81a524af7/subcommand/injector/command.go#L237-L240

And that makes its way back to os.Exit in main.go:
https://github.com/hashicorp/vault-k8s/blob/b73527fa89b29ec705daf58e49b03bd81a524af7/main.go#L17-L21

Reply:
> As per the comment above, I think this log message should include the error that triggered the shutdown.

Added the error to the log message in a45b35e

Reply:
Thanks for the explanation. I guess we are relying on the fact that `server.Shutdown()` is called before `server.ListenAndServeTLS()`?